rust_src/src/cmds.rs

```diff
@@ -378,18 +375,18 @@ fn internal_self_insert(mut c: Codepoint, n: usize) -> EmacsInt {
         }
         hairy = 2;
     }
-    synt = unsafe { syntax_property(c as i32, true) };
+    synt = unsafe { syntax_property(c.val() as i32, true) };
 
     let previous_char = if current_buffer.multibyte_characters_enabled() {
-        preceding_char() as Codepoint
+        Codepoint::from(preceding_char() as u32)
```

Comment:
Double conversions here too

Reply:
I don't think there's a good way of avoiding a double here, as I still don't feel comfortable adding conversions which could fail (or saturate, I suppose).